SecurityManager/Core/Domain/SecurityManagerPrincipal.cs

```diff
@@ -85,6 +85,9 @@ public static ISecurityManagerPrincipal Current
     private readonly SubstitutionProxy? _substitutionProxy;
     private readonly SecurityPrincipal _securityPrincipal;
 
+    [NonSerialized]
+    private Lazy<SecurityClientTransactionFactory> _securityTransactionFactory;
+
```

Comment:
This design doesn't satisfy the non-serialized + lazy-init concept. 

NonSerialized fields are NULL when the type is deserialized. One approach (probably the easiest, not sure if it's the only one) is to use a doublechecked lock. 

field: 
```
[NonSerialized]
private SecurityClientTransactionFactory _securityTransactionFactory;
```

in `private ClientTransaction CreateClientTransaction ()` 
```
  if (_securityTransactionFactory == null)
  {
    lock (_lockObject)
    {
      _securityTransactionFactory = new SecurityClientTransactionFactory();
     }
  }
  _securityTransactionFactory.Create();
```

Since we don't want to carry with us a lockobject, we can also do an Interlocked instead in `private ClientTransaction CreateClientTransaction ()` 
```
  if (_securityTransactionFactory == null)
  {
      var securityTransactionFactory = new SecurityClientTransactionFactory();
      Interlocked.CompareExchange(ref _securityTransactionFactory, securityTransactionFactory, null);
  }
  _securityTransactionFactory.Create();
```
This version will on occasion create an extra Factory instance, but that's okay. We don't care about this racy overhead.

In theory we could also use a MemoryBarrier but it is more explicit this way.

Reply:
Note: in .NET 8, they make Interlocked.CompareExchange faster:
https://devblogs.microsoft.com/dotnet/performance-improvements-in-net-8/#regex

Reply:
@a-ctor FYI missed a serialization bug.

Reply:
@Frazerus just FYI: I managed to mangle to double-checked lock pattern in the first example. It goes `if value == null` then `lock` followed by `if value == null` then `value = new`
That's why it's "double-checked" lock.